### ISSUE TITLE
ubyte format requires 0 <= number <= 255

### ISSUE THREAD

**Issue by Mahlzeit2000** [ftrobopy/ftrobopy#26]
Hi!

I'm running into an error when using ftrobopy on the txt controller with a bt remote.
my code is:

```python

import time
import ftrobopy
import threading

txt=ftrobopy.ftrobopy('auto')

joystick1 = txt.joystick(0,1,1)
joystick2 = txt.joystick(1,1,1)
Motor1 = txt.motor(1)
Motor1.setDistance(0)
Motor2 = txt.motor(2)
Motor2.setDistance(0)
txt.updateWait()

def infiniteloop1():  # this controls Motor1 and Motor2 via a BT-Remote
    while True:
        Motor1.setSpeed(joystick1.updown() * 512)
        Motor2.setSpeed(joystick1.updown() * 512)

thread1 = threading.Thread(target=infiniteloop1)
thread1.start()
```

and it returns this error

![image](https://user-images.githubusercontent.com/71544778/93662215-dd9ac900-fa5e-11ea-8075-5692502a3532.png)

Positive values going forward work fine. While going backwards with negative values its working for like 1-2 seconds and then stops with the above error. The error occurs on both cfw default and beta version ftrobopy.

Whats weird is that it works fine with leftright() but not updown():

ie this is perfectly fine in both directions:

```python
Motor1.setSpeed(joystick1.leftright() * 512)
Motor2.setSpeed(joystick1.leftright() * 512)
```
instead of

```python
        Motor1.setSpeed(joystick1.updown() * 512)
        Motor2.setSpeed(joystick1.updown() * 512)
```
which causes the error as soon as you go backwards.

EDIT: Just did some trial and error: it works fine until motor setspeed 508 after which it will throw the above error (only if resulting motor speed is negative value)

so these always work fine:
```
            Motor1.setSpeed(joystick1.updown() * 508)
            Motor2.setSpeed(joystick1.updown() * -508)
```

**Comment by ftrobopy** [comment]
Hi,

this looks strange and I have never observed this behaviour ... it could be a locking problem (?).

You could try to insert "txt.updateWait(0.01)" into the while loop to make sure that communication between the Linux-System and the Motor-Control-System of the TXT has finished. You could also try longer times than 0.01.

It will take a few days until I can test this behaviour myself ... I will come back to this.

Best wishes,
Torsten

**Comment by Mahlzeit2000** [comment]
Hi Torsten!

Thanks for getting back to me. I was able to resolve this, theres no bug in ftrobopy afterall.

The back story is that I did run across this "problem" again when I linked motor speed values to  certain PID outputs (not using threads this time). It then dawned on me that the problem is that both my PID controller as well as the BT Remote data was outputting floats like 0,xxxxx which I then simply multiplied and subsequently fed directly into setSpeed(x) without turning them into integer first. FTrobopy manual on the otherhand clearly says:

Einstellung der Motorgeschwindigkeit
      :param speed: 
      :type speed: **integer**

and when I changed my values to integer the problem was gone. So I guess I should have properly read the manual first :-)  

Best regards 

Dennis
